edsdme/blocks/dme-form/dme-form.js

```diff
@@ -0,0 +1,483 @@
+    const fieldWrapper = document.createElement('div');
+    const style = fd.Style ? ` form-${fd.Style}` : '';
+    const fieldId = `form-${fd.Type}-wrapper${style}`;
+    fieldWrapper.className = fieldId;
+    fieldWrapper.id = `${fieldId}-${idx}`;
+    fieldWrapper.classList.add('field-wrapper', 'row');
+    switch (fd.Type) {
+      case 'select':
+        fieldWrapper.classList.add('dropdownlist');
+        fieldWrapper.append(createLabel(fd));
+        fieldWrapper.append(createSelect(fd));
+        fieldWrapper.append(createErrorBlock(fd));
+        break;
+      case 'radio':
+        fieldWrapper.append(createLabel(fd));
+        fieldWrapper.append(createRadio(fd));
+        break;
+      case 'heading':
+        fieldWrapper.append(createHeading(fd, 'h3'));
+        break;
+      case 'legal':
+        fieldWrapper.append(createHeading(fd, 'p'));
+        break;
+      case 'checkbox':
+        fieldWrapper.classList.add('checkbox');
+        fieldWrapper.append(createInput(fd));
+        fieldWrapper.append(createLabel(fd));
+        break;
+      case 'text-area':
+        fieldWrapper.classList.add('textarea');
+        fieldWrapper.append(createLabel(fd));
+        fieldWrapper.append(createTextArea(fd));
+        break;
+      case 'submit':
+        fieldWrapper.classList.add('submit');
+        if (disclaimer) {
+          fieldWrapper.append(createDisclaimer(disclaimer));
+        }
+        fieldWrapper.append(createButton(fd));
+        break;
+      default: {
+        const input = createInput(fd);
+
+        fieldWrapper.classList.add('textfield');
+        fieldWrapper.append(createLabel(fd));
+        fieldWrapper.append(input);
+        fieldWrapper.append(createErrorBlock(fd));
+
+        if (fd.Placeholder) {
+          resolvePlaceholderFromProfile(fd.Placeholder).then((resolved) => {
+            if (resolved) {
+              input.setAttribute('placeholder', resolved);
+              input.value = resolved;
+            }
+          });
+        }
+        break;
+      }
+    }
+
+    if (fd.Rules) {
+      try {
+        rules.push({ fieldId: fieldWrapper.id, rule: JSON.parse(fd.Rules) });
+      } catch (e) {
+        // eslint-disable-next-line no-console
+        console.warn(`Invalid Rule ${fd.Rules}: ${e}`);
+      }
+    }
+    if (fd.RenderCondition) {
+      branching.push({ fieldId: fieldWrapper.id, condition: fd.RenderCondition });
+    }
+    form.append(fieldWrapper);
+  });
+
+  form.addEventListener('change', () => applyRules(form, rules));
+  applyRules(form, rules);
+  form.addEventListener('change', (e) => {
+    applyBranching(form, branching);
+    validate(form, false, e);
+  });
+  applyBranching(form, branching);
+  form.addEventListener('submit', (e) => {
+    e.preventDefault();
+    e.stopPropagation();
+  });
+  fill(form);
+  return (form);
+}
+
+export default async function init(el) {
+  const form = document.querySelector('.dme-form a[href$="form-definition.json"]');
```

Comment:
I think we should query "el" here instead of "document". 
not sure if it makes sense in this case, but if we need to render two forms on one page, this is important.